models/issues/issue_list.go

```diff
@@ -599,3 +599,50 @@ func (issues IssueList) GetApprovalCounts(ctx context.Context) (map[int64][]*Rev
 
 	return approvalCountMap, nil
 }
+
+// UserAssignedIssueCond return user as assignee issues list
+func UserAssignedIssueCond(userID int64) builder.Cond {
+	return builder.And(
+		builder.Eq{
+			"repository.is_private": false,
+		},
+		builder.In("issue.id",
+			builder.Select("issue_assignees.issue_id").From("issue_assignees").
+				Where(builder.Eq{
+					"issue_assignees.assignee_id": userID,
+				}),
+		),
+	)
+}
+
+// UserMentionedIssueCond return user mentioned issues list
+func UserMentionedIssueCond(userID int64) builder.Cond {
+	return builder.And(
+		builder.Eq{
+			"repository.is_private": false,
+		},
+		builder.In("issue.id",
+			builder.Select("issue_user.issue_id").From("issue_user").
+				Where(builder.Eq{
+					"issue_user.is_mentioned": true,
+					"issue_user.uid":          userID,
+				}),
+		),
+	)
+}
+
+// UserCreateIssueCond return user created issues list
+func UserCreateIssueCond(userID int64, isPull bool) builder.Cond {
+	return builder.And(
+		builder.Eq{
+			"repository.is_private": false,
+		},
+		builder.In("issue.id",
+			builder.Select("issue.id").From("issue").
+				Where(builder.Eq{
+					"issue.poster_id": userID,
+					"issue.is_pull":   isPull,
+				}),
+		),
+	)
+}
```

Comment:
What's the difference between these functions and the ones in `repo_model`? Maybe outline in a comment the differences?

Reply:
The functions in `repo_model` are only used here (but will cause the issue). So we can remove them or move these new functions into `repo_model`. I didn't do that because I'm not sure whether the functions in `repo_model` are only designed for `issuePullAccessibleRepoCond`.

The difference is that the functions in `repo_model` use `issue.repo_id in repo_ids` to filter the issue, so all issues in these repos will be caught.
So we need use `issue.id in issue_ids` to get the correct issues.